### PR TITLE
feat: implementation of pre commit hooks

### LINKED ISSUE #17: Implement pre commit hooks
**Is your feature request related to a problem? Please describe.**
Add pre commit hooks. It help catch issues like linting errors, unformatted code, or failing tests before code is committed to the repository. 

1. Linting: Ensure that code follows the defined linting rules.
2. Formatting: Automatically format code using tools like Prettier.
3. Unit Tests (when we will have) : Run unit tests to ensure no breaking changes are committed.
4. Type Checking (Optional): Add type checking if using TypeScript or similar tools.
5. Git Commit Message Formatting (Optional): Enforce a standardized format for commit messages, e.g., following Conventional Commits.

**Describe the solution you'd like**
1. on PR raise, pre commit hook should run and throw the errors on failing

2. Code cannot be committed if linting, formatting, or tests fail.

**Describe alternatives you've considered**
NA

**Additional context**
NA

### PR DESCRIPTION
**Pull request by Neha**
# Description

Added husky pre-commit hook for:

1. eslint 
2. prettier
3. commit messages

Fixes #17 

**Snapshots/Videos:**
Screenshot 2024-10-29 at 21 50 07
Screenshot 2024-10-29 at 21 49 50

## Type of change

- 🍕 Feature

## Added tests?

- 🙅 No, because they aren't needed

## Internationalization Support?

- 🙅 No, because they aren't needed

**Steps to test the feature:**
1. run the project at local
2. do some changes
3. try to commit the changes without proper message.
4. observe, this should throw an error
5. do some formatting issues 
6. commit the changes
7. observe, the formatting issues will be automatically fixed

**Snapshot of the test-cases that are passing:**
NA

**Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
NA

"By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice."

### CODE REVIEW

**Comment by giolaq** [comment]
Thank you! 

Can you expand on the rules used and why are they useful? It's good to have this for the future .

Also why using husky? Can we use git hooks? What's the benefit of it?

**Comment by Neha** [comment]
> Thank you!
> 
> Can you expand on the rules used and why are they useful? It's good to have this for the future .
> 
> Also why using husky? Can we use git hooks? What's the benefit of it?

prettier rules are the standard rules for managing the use of tabs, width, singlequotes, etc. In , `pettierrc` these rules are mentioned. As required in future we can add, or edit rules. These are useful to manage the consistency of the project's code formatting. In the absence of this, ever developer will be formatting the code as per their preference  and this will leads to inconsistency, and in PR lot of unnecessary diff will get logged

ESLint rules have the react, typescript, react-native, and jsx-a11y. The few rules to start with are - file extension should be .tsx, no-inline-style, no unused var, etc. These rule are mentioned in file `eslint.config.js`. These rules can be edited in future. ESLint rules will make sure that developers are using the right and good pattern of coding as suggested by the React, TS, and RN. While coding if there would be any issue, ESLint will highlight. Eg: if the accessibility is missing , or a variable is not used but declared then while coding eslint will throw error or warning as per the rule and help the developer to do the needful. 

Husky is flexible and simpler than git. With husky we just need package.json to setup, and commands. However with git the pre commit hooks installation is not easier. Another bigger advantage of husky over git is, husky will be installed for every developer however with git, every developer has to setup on their machine. So, all environments will be handle by the husky as compare to the git

File: components/MenuContext.tsx [hunk 1]
```diff
@@ -10,16 +10,15 @@ const MenuContext = createContext<MenuContextType | undefined>(undefined);
...
-    isOpen,
-    toggleMenu: setIsOpen,
-  }), [isOpen]);
-
-  return (
-    <MenuContext.Provider value={value}>
-      {children}
-    </MenuContext.Provider>
+  const value = useMemo(
+    () => ({
```
[the remark this answers is not in the document]
Comment (Neha) [reply components/MenuContext.tsx:14]: this is formatting test. We can keep this file.

File: yarn.lock
[the remark this answers is not in the document]
Comment (Neha) [reply yarn.lock:1]: yes, it is yarn.lock we do need this for fix version packages

**Review by Neha — COMMENTED** [review]
all changes are done. Please re-review

File: yarn.lock
[the remark this answers is not in the document]
Comment (giolaq) [reply yarn.lock:1]: So are you proposing to use yarn instead of npm?

**Comment by Neha** [comment]
@giolaq as suggested, changes are done. Please recheck. Thanks

File: components/CustomDrawerContent.tsx [hunk 2]
```diff
@@ -19,7 +19,10 @@ export default function CustomDrawerContent(props: any) {
 
   return (
     <SpatialNavigationRoot isActive={isMenuOpen}>
-      <DrawerContentScrollView {...props} style={styles.container} scrollEnabled={false}>
+      <DrawerContentScrollView {...props} style={styles.container} scrollEnabled={false} contentContainerStyle={{
```
Comment (giolaq) [inline components/CustomDrawerContent.tsx:22]: This change is already merged, make sure you update to the latest main

**Review by giolaq — COMMENTED** [review]
part related to the pre-hooks and eslinting looks good.
Make sure you get the changes in main

**Review by giolaq — APPROVED** [review]
LGTM

**Pull request merged**